packages/roosterjs-content-model-core/lib/corePlugin/entity/entityDelimiterUtils.ts

```diff
@@ -303,3 +320,31 @@ export const handleEnterInlineEntity: ContentModelFormatter = model => {
 
     return true;
 };
+
+const triggerEntityEventOnEnter = (
+    editor: IEditor,
+    wrapper: HTMLElement,
+    rawEvent: KeyboardEvent
+) => {
+    const format = parseEntityFormat(wrapper);
+    if (format.id && format.entityType && !format.isFakeEntity) {
+        editor.triggerEvent('entityOperation', {
+            operation: 'click',
+            entity: {
+                id: format.id,
+                type: format.entityType,
+                isReadonly: !!format.isReadonly,
+                wrapper,
+            },
+            rawEvent: rawEvent,
+        });
+    }
+};
+
+const getSelectedEntity = (selection: RangeSelection) => {
```

Comment:
In PR https://github.com/microsoft/roosterjs/pull/2492, @florian-msft  is adding a method to get current entity from a given node. I think you can wait and leverage the function

Reply:
![image](https://github.com/microsoft/roosterjs/assets/23065085/abc3e6f3-ab1e-4d75-a57a-6a3ae4b22116)

Reply:
@juliaroldi the PR is merged, you can use the new API now.